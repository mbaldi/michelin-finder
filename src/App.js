import React, { useState } from 'react';
import axios from 'axios';
import styled from 'styled-components';
import { Typography, Divider } from 'antd';

import Autocomplete from './components/Autocomplete';
import Gallery from './components/Gallery';

const API_URL =
  process.env.REACT_APP_API_URL !== '<REACT_APP_API_URL>'
    ? process.env.REACT_APP_API_URL
    : 'http://localhost:1337/api';

export const client = axios.create({
  baseURL: API_URL,
  headers: {
    'Content-Type': 'application/json',
  },
});

const AppWrapper = styled.div`
  text-align: center;
  background-color: #fcfbfc;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const { Title } = Typography;

const App = () => {
  const [location, setLocation] = useState();

  const handleAutoCompleteClick = async (address) => {
    //geocode address
    const { data } = await client.post('/geocode', {
      address,
    });
    setLocation(data);
  };

  return (
    <AppWrapper>
      <Title level={3}>
        Welcome! we will find Michellin-rated restaurants near you
      </Title>
      <Autocomplete
        placeholder="Please type your address or location"
        onClick={handleAutoCompleteClick}
      />
      <Divider />
      {location && <Gallery location={location}></Gallery>}
    </AppWrapper>
  );
};

export default App;
