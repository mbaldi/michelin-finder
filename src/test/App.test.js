import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('App', () => {
  beforeEach(() => {
    global.google = {
      maps: {
        places: {
          Autocomplete: jest.fn().mockImplementation(() => {
            const setFields = jest.fn();
            const addListener = jest.fn();
            return { setFields, addListener };
          }),
        },
      },
    };
  });

  it('renders component', () => {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });
});
