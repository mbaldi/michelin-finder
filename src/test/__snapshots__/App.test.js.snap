// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component 1`] = `
<styled.div>
  <Title
    level={3}
  >
    Welcome! we will find Michellin-rated restaurants near you
  </Title>
  <Autocomplete
    onClick={[Function]}
    placeholder="Please type your address or location"
  />
  <Divider />
</styled.div>
`;
